### PR TITLE
Add ConfirmButton component

### PR DESCRIPTION
**Pull request by cwardzala**

### CODE REVIEW

**Review by nsmedira — COMMENTED** [review]
reviewed 2022-07-22 at 12:14 ET

File: src/components/ConfirmButton.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    };
+
+    const handleClick = () => {
+        switch (buttonState) {
+            case CONFIRM_STATES.CONFIRM:
+                setButtonState(CONFIRM_STATES.DONE);
+                reset();
+                onClick();
+                break;
+            case CONFIRM_STATES.READY:
```
[the remark this answers is not in the document]
Comment (cwardzala) [reply src/components/ConfirmButton.jsx:59]: Its combined with `default:`

File: src/pages/Components/ButtonsPage.jsx [hunk 2]
```diff
@@ -175,6 +176,34 @@ const ButtonsPage = () => {
...
+                    confirmClass="btn-yellow"
+                />
+            </p>
+            <p className="mb-8">
+                <ConfirmButton label="Custom 10s Timeout" timeout={10000} />
+            </p>
+            <Code
+                className="mb-6"
+                lang="jsx"
+                code={`<ConfirmButton />\n<ConfirmButton label="Do something destructive!" confirmLabel="Are you absolutely sure, this cannot be un-done?" successLabel="Yay! You destroyed all the puny humans!"/>\n<ConfirmButton label="Custom Classes" defaultClass="btn-fuchsia" successClass="btn-lime" confirmClass="btn-yellow"/>\n<ConfirmButton label="Custom 10s Timeout" timeout={10000} />`}
```
[the remark this answers is not in the document]
Comment (cwardzala) [reply src/pages/Components/ButtonsPage.jsx:204]: Whats so funny?

[the remark this answers is not in the document]
Comment (nsmedira) [reply src/pages/Components/ButtonsPage.jsx]: "Yay! You destroyed all the puny humans!"

File: src/components/ConfirmButton.jsx [hunk 3]
```diff
@@ -22,17 +23,17 @@ const ConfirmButton = ({
...
 
-    let isConfirm = buttonState === CONFIRM_STATES.CONFIRM || buttonState === CONFIRM_STATES.DONE;
+    const isConfirm = [CONFIRM_STATES.CONFIRM, CONFIRM_STATES.DONE].includes(buttonState);
 
     const handleTimeout = () => {
         setButtonState(CONFIRM_STATES.READY);
     };
 
-    const [isReady, cancel, reset] = useTimeoutFn(handleTimeout, isConfirm ? timeout : false); // eslint-disable-line
+    const [, , reset] = useTimeoutFn(handleTimeout, isConfirm ? timeout : false);
```
Comment (nsmedira) [inline src/components/ConfirmButton.jsx:36]: this is cool

File: src/components/ProgressBar.jsx [hunk 4]
```diff
@@ -3,12 +3,13 @@ import classNames from 'classnames';
 import PropTypes from 'prop-types';
 
 const ProgressBar = ({ value, label, max, className = '', barClassName = '' }) => {
-    let progress = value >= 0 ? Math.ceil((value / max) * 100) : undefined;
-    console.log(progress);
+    let hasValue = ![null, undefined, false].includes(value);
+    let progress = hasValue ? Math.ceil((value / max) * 100) : undefined;
```
Comment (nsmedira) [inline src/components/ProgressBar.jsx:7]: this seems to change the logic such that now `progress` can be negative. maybe

```jsx
const hasValue = typeof value === 'number' && value >= 0;
```

?

File: src/pages/Components/ButtonsPage.jsx [hunk 5]
```diff
@@ -178,13 +178,25 @@ const ButtonsPage = () => {
...
+                    }}
+                />
             </p>
             <p>
                 <ConfirmButton
                     label="Do something destructive!"
                     confirmLabel="Are you absolutely sure, this cannot be un-done?"
                     successLabel="Yay! You destroyed all the puny humans!"
+                    onConfirm={() => {
+                        console.log('Done!');
```
Comment (nsmedira) [inline src/pages/Components/ButtonsPage.jsx:198]: is log intentional?

**Review by nsmedira — COMMENTED** [review]
reviewed 2022-08-01 at 10:05 ET

File: src/pages/Components/ButtonsPage.jsx (hunk 5 above)
Comment (cwardzala) [reply src/pages/Components/ButtonsPage.jsx:198]: Yes, this is just the demo page and we need to include an `onConfirm`.

**Review by nsmedira — APPROVED** [review]
reviewed and approved 2022-08-01 at 10:29 ET

**Pull request merged**